nuxeo-jsf/nuxeo-platform-webapp-base/src/main/java/org/nuxeo/ecm/webapp/locale/LocaleStartup.java

```diff
@@ -108,18 +109,53 @@ public void setupLocale(DocumentModel userProfileDoc) {
     }
 
     protected void setupLocale(Locale locale) {
-        if (locale == null) {
-            log.debug("Locale not set, falling back to request locale");
-            locale = FacesContext.getCurrentInstance().getExternalContext().getRequestLocale();
+        FacesContext ctx = FacesContext.getCurrentInstance();
+        if (locale == null && ctx != null) {
+            if (log.isDebugEnabled()) {
```

Comment:
As there's no parameters in message, I think we don't need to test if debug is enabled.
l.114 l.120 l.126